src/pages/UpdateServers.js

```diff
@@ -0,0 +1,198 @@
+// (c) Copyright 2017-2018 SUSE LLC
+/**
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+import React from 'react';
+import { translate } from '../localization/localize.js';
+import CollapsibleTable from '../components/CollapsibleTable.js';
+import { ActionButton } from '../components/Buttons.js';
+import { updateServersInModel, getMergedServer } from '../utils/ModelUtils.js';
+import { MODEL_SERVER_PROPS_ALL, MODEL_SERVER_PROPS } from '../utils/constants.js';
+import { fetchJson, putJson } from '../utils/RestUtils.js';
+import { UpdateServerPages } from './ReplaceServer/UpdateServerPages.js';
+import BaseWizardPage from './BaseWizardPage.js';
+
+class UpdateServers extends BaseWizardPage {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      model: this.props.model
+    };
+  }
+
+  componentWillReceiveProps(newProps) {
+    //deal with timing issue when load model
+    this.setState({
+      model : newProps.model,
+    } , () => {
```

Comment:
As mentioned above, calling `this.setState` in the callback to `this.setState` is problematic and should be avoided.

Reply:
will update